### PR TITLE
Add global friend_connection status callback, for use with conference rejoining

### PR DESCRIPTION
**Pull request by zugz**
This lets any friend connection, whether from messenger or from conferences, 
to be used for rejoining a conference. This is how it was meant to work all 
along.

I've also included a couple of tweaks to the conference test in this pr to 
make it more robust. Hopefully, and as far as my (limited) testing can 
determine, the conference test should reliably pass with this pr.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1189)

### CODE REVIEW

File: toxcore/group.c [hunk 1]
```diff
@@ -878,13 +878,23 @@ static void rejoin_frozen_friend(Group_Chats *g_c, int friendcon_id)
     }
 }
 
+static int g_handle_any_status(void *object, int friendcon_id, uint8_t status, void *userdata)
+{
+    Group_Chats *g_c = (Group_Chats *)object;
+
+    if (status) {
```
Comment (iphydf) [inline toxcore/group.c]: `if (status != ...something?)` some constant here? I think this is an enumeration in practice.

File: toxcore/friend_connection.c [hunk 2]
```diff
@@ -403,6 +406,10 @@ static int handle_status(void *object, int number, uint8_t status, void *userdat
     if (status_changed) {
         unsigned int i;
```
Comment (iphydf) [inline toxcore/friend_connection.c]: Move `i` below the `if` or even into the `for` init decl.

File: toxcore/group.c (hunk 1 above)
Comment (zugz) [reply toxcore/group.c]: It's not an enum, just 0 or 1. Presumably we'll make it a bool one day. So "(status)" seems correct.

File: toxcore/friend_connection.c (hunk 2 above)
Comment (zugz) [reply toxcore/friend_connection.c]: Agreed, but this has nothing to do with this PR.

**Comment by zugz** [comment]
I'm closing this PR. It was based on a misdiagnosis of a bug. This callback isn't needed until we have saving, so there's no need to cherry-pick it out as a separate PR. I'll submit a new PR soon with an actual fix to the bug and the test changes of this PR. Nothing to see here.

File: toxcore/friend_connection.c (hunk 2 above)
Comment (iphydf) [reply toxcore/friend_connection.c]: It does, because you introduce new lines between the declaration and the `for`.

**Comment by zugz** [comment]
* Thursday, 2018-09-20 at 17:41 +0000 - iphydf <notifications@github.com>:

>It does, because you introduce new lines between the declaration and 
>the `for`.

Ah yes, I see. Thanks. Fixed in the saving PR.

**Pull request closed without merging**